src/workerd/server/tests/python/python-rpc/worker.js

```diff
@@ -0,0 +1,59 @@
+import { WorkerEntrypoint } from 'cloudflare:workers';
+
+import * as assert from 'node:assert';
+
+export class JsRpcTester extends WorkerEntrypoint {
+  async noArgs() {
+    return 'hello from js';
+  }
+  async oneArg(a) {
+    return `${a}`;
+  }
+  async identity(x) {
+    return x;
+  }
+  async handleResponse(resp) {
+    // Verify that we receive a JS object here...
+    assert.deepStrictEqual(resp.constructor.name, 'Response');
+    return resp;
+  }
+
+  async handleRequest(req) {
+    assert.deepStrictEqual(req.constructor.name, 'Request');
+    return req;
+  }
+}
+
+export default {
+  async test(ctrl, env, ctx) {
+    // JS types
+    for (const val of [
+      1,
+      'test',
+      [1, 2, 3],
+      new Map([['key', 42]]),
+      42,
+      1.2345,
+      false,
+      true,
+      undefined,
+    ]) {
+      const response = await env.PythonRpc.identity(val);
+      assert.deepStrictEqual(response, val);
+    }
+
+    const null_resp = await env.PythonRpc.identity(null);
+    assert.deepStrictEqual(null_resp, undefined);
+
+    // Web/API Types
+    const py_response = await env.PythonRpc.handle_response(
+      new Response('this is a response')
+    );
+    assert.deepStrictEqual(await py_response.text(), 'this is a response');
```

Comment:
This should be a js Response object but the test doesn't seem to make this clear

Reply:
I can add some checks to make it clear, but it definitely will be a JS response, there is no other way this can work (you cannot send a PyProxy through the RPC)